apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClientImpl.cs

```diff
@@ -196,12 +203,26 @@ public override IList<Language> ListLanguages(string target = null)
         }
 
         /// <inheritdoc />
-        public override async Task<IList<Detection>> DetectLanguageAsync(string text, CancellationToken cancellationToken = default(CancellationToken))
+        public override async Task<Detection> DetectLanguageAsync(string text, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            GaxPreconditions.CheckNotNull(text, nameof(text));
```

Comment:
Why not a one-liner, as in the sync case?
And this is delegating, so could be in the base abstract class.

If keeping like this, maybe at least combine the next two lines. It looks rather odd having the task in its own local!